### PR TITLE
Add selenium testing

### PR DESCRIPTION
**Pull request by JackMead**
Tests pass using the commands mentioned in the ReadMe. As there's no front end functionality on master at time of branching, the tests are basically placeholders. Once there's a login form I'll have a go at extending these to actually do something!

Testing the Jenkin's job to check it runs fine at the moment. Will only be able to test it fully once this is through to master since the target.address parameter obviously has no effect currently.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,4 +71,10 @@ To run the server side unit tests in intelliJ, right click on the `AgentDiscover
...
 ## Frontend Tests
-To run the front-end unit tests, from command line navigate to `AgentDiscoveries-Frontend` and then run npm test.
\ No newline at end of file
+To run the front-end unit tests, from command line navigate to `AgentDiscoveries-Frontend` and then run npm test.
+
+## End-to-End Tests
+In order to run the End-To-End tests locally the webserver needs to be active, so ensure you first run the application as described in the "Run" section.
+
+Once the server is online, you can run the tests from the command line using `mvn verify -P integration-test`. This can be run from either the root directory or AgentDiscoveries-Backend. It will target the address `http://localhost:8080` by default.
+If you need to run the tests on another address for any reason, use the command `mvn verify -P integration-test -Dtarget.Address="http://foo.bar"` with the relevant address.
```
Comment (AnthonyPengelly) [inline README.md]: Here you have a capital A for address, whereas it's lowercase in the pom. Any idea if that matters?

File: AgentDiscoveries-Backend/pom.xml [hunk 2]
```diff
@@ -221,6 +260,25 @@
...
+                    <systemPropertyVariables>
+                        <target.address>${target.address}</target.address>
+                    </systemPropertyVariables>
+                </configuration>
+                <version>2.20.1</version>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>integration-test</goal>
+                            <goal>verify</goal>
```
Comment (AnthonyPengelly) [inline AgentDiscoveries-Backend/pom.xml]: How do these goals cause your selenium tests to be run? 😛 There's some magic here somewhere

File: AgentDiscoveries-Backend/src/test/java/org/softwire/training/api/endToEnd/dummySeleniumIT.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Test
+    public void testTitle(){
+        WebDriver driver = new HtmlUnitDriver();
+        driver.get(Target_Address);
+        String title = driver.getTitle();
+        assertTrue(title.equals("Agent Discoveries"));
+    }
+
+    public void testOnline(){
```
Comment (AnthonyPengelly) [inline AgentDiscoveries-Backend/src/test/java/org/softwire/training/api/endToEnd/dummySeleniumIT.java]: Does this need a `@Test` attribute?

File: README.md (hunk 1 above)
Comment (JackMead) [reply README.md]: Good catch, it needs to be consistent yes!

File: AgentDiscoveries-Backend/pom.xml [hunk 4]
```diff
@@ -23,6 +23,28 @@
         <mainClass>org.softwire.training.AgentDiscoveriesApplication</mainClass>
     </properties>
 
+    <profiles>
+        <profile>
+            <id>integration-test</id>
+            <properties>
+                <skip.unit.tests>true</skip.unit.tests>
+                <skip.integration.tests>false</skip.integration.tests>
+                <target.address>http://localhost:8080</target.address>
```
[the remark this answers is not in the document]
Comment (JackMead) [reply AgentDiscoveries-Backend/pom.xml]: Yeah exactly that, if you provide a parameter it automatically takes precedence over any set in the pom apparently :)

File: AgentDiscoveries-Backend/pom.xml (hunk 2 above)
Comment (JackMead) [reply AgentDiscoveries-Backend/pom.xml]: The goals tells it to use the maven-failsafe-plugin for the integration-test step, in which it runs all tests of forms like /IT*.java or /*IT.java
...Now I look at it again, how does it know how to run the unit tests? 😕

File: AgentDiscoveries-Backend/src/test/java/org/softwire/training/api/endToEnd/dummySeleniumIT.java (hunk 3 above)
Comment (JackMead) [reply AgentDiscoveries-Backend/src/test/java/org/softwire/training/api/endToEnd/dummySeleniumIT.java]: Ermm, apparently not as it was definitely running it anyway! But should definitely be there for clarity haha

**Pull request merged**